src/main/java/frc/robot/subsystems/intake/commands/IntakeByRobotSpeed.java

```diff
@@ -0,0 +1,43 @@
+package frc.robot.subsystems.intake.commands;
+
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.intake.Intake;
+
+import java.util.function.DoubleSupplier;
+
+/**
+ * intakes balls according to robot speed.
+ */
+public class IntakeByRobotSpeed extends CommandBase {
+    private final Intake intake;
+    private final DoubleSupplier robotSpeedMeterPerSecond;
+
+    public IntakeByRobotSpeed(Intake intake, DoubleSupplier speed) {

```

Comment:
You should rename `speed` to `robotSpeedMeterPerSeconds`. First, to match the class' member name, and second, to clarify this is not the speed of the intake we're talking about.